templates/components/itilobject/timeline/knowledge_item.html.twig

```diff
@@ -0,0 +1,81 @@
+{#
+ # ---------------------------------------------------------------------
+ #
+ # GLPI - Gestionnaire Libre de Parc Informatique
+ #
+ # http://glpi-project.org
+ #
+ # @copyright 2015-2025 Teclib' and contributors.
+ # @copyright 2003-2014 by the INDEPNET Development Team.
+ # @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ #
+ # ---------------------------------------------------------------------
+ #
+ # LICENSE
+ #
+ # This file is part of GLPI.
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+{% import 'components/form/fields_macros.html.twig' as fields %}
+
+{% set search_button %}
+    <button type="button" class="btn btn-secondary overflow-hidden text-nowrap" name="search_{{ itilobject }}"
+        title="{{ __('Search a ' ~ itilobject) }}" data-bs-toggle="tooltip" data-bs-placement="top">
+    <i class="ti ti-search"></i>
+    </button>
+{% endset %}
+{{ fields.field(
+        '',
+        search_button,
+        '',
+        {
+            'full_width': true,
+            'icon_label': true,
+        }
+    )
+}}
+
+<script>
+    $('button[name="search_{{ itilobject }}"]').on('click', () => {
+    window.glpi_ajax_dialog({
+        id: 'modal_searchItilobject',
+        modalclass: 'modal-xl',
+        title: '{{ __('Search a ' ~ itilobject) }}',
+        url: '{{ path('/Knowbase/KnowbaseItem/Search/') ~ itilobject ~ '/' ~ (item.getType() ~ '/' ~ item.getID())|e('js') }}'
+    });
+    $(document).off('click', '#modal_searchItilobject button.use-itilobject').on('click', '#modal_searchItilobject button.use-itilobject', (e) => {
+        const btn = $(e.target).closest('button.use-itilobject');
+        const itilobject_id = btn.attr('data-itilobject-id') ?? btn.closest('.list-group-item').attr('data-itilobject-id');
```

Comment:
This variable is not named properly, indeed, the controller expect this to be a `knowbaseitems_id`.